src/utils.php

```diff
@@ -645,3 +645,87 @@ function ensure_dir( $dir ) {
 
 	return realpath( $dir );
 }
+
+/**
+ * Recursively check permissions in a directory and confirm the UID/GID is properly set.
+ *
+ * @param string $dir The directory path to check.
+ *
+ * @return string|bool Error message or true if all checks pass.
+ */
+function check_permissions_in_directory_recursively( string $dir ) {
+	// Open the directory.
+	if ( $handle = opendir( $dir ) ) {
+		// Loop through each entry in the directory.
+		while ( false !== ( $entry = readdir( $handle ) ) ) {
+			// Skip '.' and '..' directories.
+			if ( $entry === '.' || $entry === '..' ) {
+				continue;
+			}
+
+			// Construct the full file path.
+			$file_path = $dir . '/' . $entry;
+
+			// Define the invalid ID and the error message.
+			$invalid_id = 100999;
```

Comment:
No need to redefine a fixed message in each iteration: prepare a `sprintf` template outside of the loop and compile it when and if required.